lib/bolt/result.rb

```diff
@@ -73,6 +73,24 @@ def self.from_asserted_args(target, value)
       new(target, value: value)
     end
 
+    def self._pcore_init_from_hash
+      raise "Result shouldn't be instantiated from a pcore_init class method. How did this get called?"
+    end
+
+    def _pcore_init_from_hash(init_hash)
+      opts = init_hash.reject { |k, _v| k == 'target' }
+      initialize(init_hash['target'], opts.transform_keys(&:to_sym))
```

Comment:
We can't use `transform_keys` in Ruby 2.3. Use `Bolt::Util.walk_keys(opts, &:to_sym)` instead. This is also the case on `ApplyResult`.